### PR TITLE
Update to use aws-sdk-go-v2

### PR DESCRIPTION
**Pull request by harlow**
This is sweeping update of the client library to use the V2 Golang implementation of AWS SDK

Fixes: #138

- [x] Upgrade library to AWS SDK V2
- [x] Go forward plan for de-aggregation (not included in v2 sdk)
- [ ] Go forward plan for mocking/stubbing (v2 sdk has removed the iface)
- [x] Run end-to-end tests against local kinesis
- [ ] Run end-to-end tests against production kinesis

Blocked on: https://github.com/awslabs/kinesis-aggregation/pull/143

### CODE REVIEW

File: allgroup.go [hunk 1]
```diff
@@ -5,16 +5,15 @@ import (
...
 
-	"github.com/aws/aws-sdk-go/service/kinesis"
-	"github.com/aws/aws-sdk-go/service/kinesis/kinesisiface"
+	"github.com/aws/aws-sdk-go-v2/service/kinesis/types"
 )
 
 // NewAllGroup returns an intitialized AllGroup for consuming
 // all shards on a stream
-func NewAllGroup(ksis kinesisiface.KinesisAPI, store Store, streamName string, logger Logger) *AllGroup {
+func NewAllGroup(ksis kinesisClient, store Store, streamName string, logger Logger) *AllGroup {
```
Comment (harlow) [inline allgroup.go:13]: it looks like V2 removed the `kinesisiface.KinesisAPI` but I didn't want to give up the ability to mock out the client so I created a local interface for `kinesisClient`, not ideal but should work for time being while they figure out what to do about KinesisAPI

https://github.com/aws/aws-sdk-go-v2/issues/70

File: consumer.go [hunk 2]
```diff
@@ -183,17 +180,19 @@ func (c *Consumer) ScanShard(ctx context.Context, shardID string, fn ScanFunc) e
...
-				if err != nil {
-					return err
-				}
-			} else {
-				records = resp.Records
-			}
-			for _, r := range records {
+			// var records []types.Record
+
+			// TODO(hw): need a aws-sdk-v2 deaggregate
```
Comment (harlow) [inline consumer.go:185]: @harlow need to figure out what to do here. Implement a local version?

**Comment by harlow** [comment]
@akrantz01 any chance you can pull this branch into yours and see if everything lines up OK? All the tests I've run locally are going well; after I get the de-aggregation sorted I think its really close

**Comment by akrantz01** [comment]
It's hard to say whether everything lines up correctly. Telegraf was on v0.3.1 before updating, and by the looks of it, a handful of things changed between v0.3.1 and v0.3.5.

Aside from updating between v0.3.1 and v0.3.5, all the tests pass without modifying any core components. So it _seems_ like everything lines up correctly, but I could be wrong. I've also yet to test actually connecting to AWS Kinesis.

**Pull request merged**